src/main/java/sirius/biz/process/Processes.java

```diff
@@ -503,6 +504,20 @@ protected boolean updateState(String processId, ProcessState newState) {
         return modify(processId, null, process -> process.setState(newState));
     }
 
+    /**
+     * Marks a process as running.
+     *
+     * @param processId the process to update
+     * @return <tt>true</tt> if the process was successfully modified, <tt>false</tt> otherwise
+     */
+    protected boolean markRunning(String processId) {
+        return modify(processId, process -> process.getState() == ProcessState.WAITING, process -> {
+            process.setStarted(LocalDateTime.now());
+            process.setWaitingTime((int) Duration.between(process.getCreated(), process.getStarted()).getSeconds());
```

Comment:
As far as I understand it, the waiting time is only set when a process is started, right? Should we also set it when canceling a process before it is started? Can we/should we distinguish between "canceled while waiting" vs "canceled while running"?

Reply:
That's a good question, on one hand it feels right to set the field also when a job is cancelled. On the other hand, it would decrease the average waiting time when analyzing without representing the real time the process would have to wait until actual execution.